Doc/howto/logging-cookbook.rst

```diff
@@ -765,13 +765,71 @@ serialization.
++-------------------------+----------------------------------------------------+
+| :file:`supervisor.conf` | The Supervisor configuration file, which has       |
+|                         | entries for the listener and a multi-process web   |
+|                         | application                                        |
++-------------------------+----------------------------------------------------+
+| :file:`ensure_app.sh`   | A Bash script to ensure that Supervisor is running |
+|                         | with the above configuration                       |
++-------------------------+----------------------------------------------------+
+| :file:`log_listener.py` | The socket listener program which receives log     |
+|                         | events and logs them to a file                     |
++-------------------------+----------------------------------------------------+
+| :file:`main.py`         | A simple web application which does logging via a  |
+|                         | socket connected to the listener                   |
++-------------------------+----------------------------------------------------+
+| :file:`webapp.json`     | A JSON configuration file for the web application  |
++-------------------------+----------------------------------------------------+
+| :file:`client.py`       | A Python script to exercise the web application    |
++-------------------------+----------------------------------------------------+
+
+The web application uses `Gunicorn <https://gunicorn.org/>`_, which is a
+popular web application server which starts multiple worker processes to handle
+requests. The whole setup shows how the workers can write to the same log file
+without treading on each other's toes --- they go through the socket listener.
+
+To test these files, do the following in a POSIX environment:
+
+1. Download `the Gist
+   <https://gist.github.com/vsajip/4b227eeec43817465ca835ca66f75e2b>`_, using
+   the "Download ZIP" button, as a :file:`zip` archive.
+
+2. Unzip the above files from the archive into a scratch directory.
+
+3. In the scratch directory, run :program:`bash prepare.sh` to get things ready.
+   This creates a :file:`run` subdirectory to contain Supervisor related and
+   log files, and a :file:`venv` subdirectory to contain a virtual environment
+   into which ``bottle``, ``gunicorn`` and ``supervisor`` are installed.
+
+4. Run :program:`bash ensure_app.bash` to ensure that Supervisor is running with
+   the above configuration.
+
+5. Run :program:`venv/bin/python client.py` to exercise the web application,
+   which will lead to records being written to the log.
+
+6. Inspect the log files in the :file:`run` subdirectory. You should see the
+   most recent log lines in files matching :file:`app.log*`. They won't be in
+   any particular order, since they have been handled concurrently by different
+   worker processes, and that does not happend in a deterministic way.
```

Comment:
```suggestion
   worker processes, and that does not happen in a deterministic way.
```

Reply:
```suggestion
   worker processes in a non-deterministic way.
```

This can be made simpler, less wordy and more readable, and also avoids the above typo